### DISCUSSION TITLE
DICOMSegmentationWriterOperator throwing ValueError

### DISCUSSION THREAD

**Discussion by ArpanGyawali** [Project-MONAI/monai-deploy-app-sdk#446] [General]
I am using my own operator for the task of Skull Stripping and deploying the application. My pipeline is
- DICOMDataLoaderOperator()
- DICOMSeriesSelectorOperator(rules=Sample_Rules_Text)
- DICOMSeriesToVolumeOperator()
- SkullStripOperator()
- DICOMSegmentationWriterOperator(segment_descriptions=segment_descriptions, custom_tags=custom_tags)

Every operations are successful and the predicted nifty file is saved but at writing dicom file operation it throws an error.
`ValueError: Number of plane positions in source image(s) does not match size of first dimension of "pixel_array" argument.`

Also my predicted nifty segmentation is flipped as shown in figure
![image](https://github.com/Project-MONAI/monai-deploy-app-sdk/assets/46037105/aae856b8-5964-444a-a88c-54bc7ad6ed43)

And i get this warning while loading the segmentation in ITKSnap
![image](https://github.com/Project-MONAI/monai-deploy-app-sdk/assets/46037105/a5d28097-2a02-4650-8dc0-5f76a75f3861)

I tried using various post transforms like Flipd and Orientationd as seen in the code below to reorient the prediction but the result is the same,

**SkullStrip_operator.py**
```
    def post_process(self, pre_transforms: Compose, out_dir: str = "./prediction_output") -> Compose:
        """Composes transforms for postprocessing the prediction results."""

        pred_key = self._pred_dataset_key
        return Compose(
            [
                EnsureTyped(keys=pred_key),
                Activationsd(keys=pred_key, softmax=True),
                AsDiscreted(keys=pred_key, argmax=True),
                KeepLargestConnectedComponentd(keys=pred_key, applied_labels=1),
                Flipd(keys=[pred_key], spatial_axis=2),
                SaveImaged(
                    keys=pred_key,
                    output_dir=out_dir,
                    output_postfix="seg",
                    output_dtype=uint8,
                ),
            ]
        )

```
This flipd transform doesnot have any effect in output. 

Is the previous error i mentioned due to this oncorrect orientation?
Any help would be very appreciated.

**Comment by MMelQin** [comment]
It is likely the DIMs are switched, i.e. HWD vs DHW, as the error indicates.

**Reply by ArpanGyawali** [reply]
Thanks,
how to correct this?

**Reply by inikishev** [reply]
I have the same issue, is there a fix? Right now I manually flip the segmentation but it is seemingly random how it gets flipped

**Comment by MMelQin** [comment]
The image looks flipped, rendered with neurology convention while the viewer is in radiology convention. I will find time to take a closer look one back from vacation in two weeks. BTW, a new version of the App SDK has been published, so the existing app based on the App SDK v0.5.1 and lower needs to pin modal-deploy-app-sdk to 0.5.1 or ~=0.5.1. Thanks.

**Reply by ArpanGyawali** [reply]
@MMelQin Have you got the time to look into this problem?
